components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -70,11 +68,50 @@ namespace brave {
 
 namespace {
 
+// NOTE TO READER: These headers are for partners to determine the browser is
+// Brave without creating webcompat issues by creating a new user-agent. Since
+// there are tens of millions of Brave users, this presents no privacy risk.
+// Custom headers are deprecated and new partners use the
+// navigator.brave.isBrave() JavaScript API.
+constexpr char kPartnerEaffName[] = "eaff";
+constexpr char kPartnerUpholdName[] = "uphold";
+constexpr char kPartnerGrammarlyName[] = "grammarly";
+
+const std::vector<std::string> kPartnerEaffDomains = {"eaff.com",
+                                                      "stg.eaff.com"};
+const std::vector<std::string> kPartnerUpholdDomains = {
+    "sandbox.uphold.com", "api-sandbox.uphold.com", "uphold.com",
+    "api.uphold.com"};
+const std::vector<std::string> kPartnerGrammarlyDomains = {
+    "www.grammarly.com", "grammarly.com", "static.grammarly.com",
+    "gnar.grammarly.com"};
+
 BraveReferralsService::ReferralInitializedCallback*
     g_testing_referral_initialized_callback = nullptr;
 
 base::FilePath g_promo_file_path;
 
+base::Value CreateReferralHeader(
+    const char partner_name[],
+    const std::vector<std::string>& partner_domains) {
+  base::Value headers_dict(base::Value::Type::DICTIONARY);
+  base::Value cookie_names(base::Value::Type::LIST);
```

Comment:
`cookieNames` is empty for all three partners. It would be nice to try and remove this dead code since we're not using it and some people might be wondering why it looks like we're doing something with cookies.

Reply:
fixed